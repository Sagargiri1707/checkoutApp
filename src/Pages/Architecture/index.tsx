const Tick = () => (
  <svg
    className="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
    aria-hidden="true"
    xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    viewBox="0 0 20 20"
  >
    <path d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z" />
  </svg>
);
const dataFlow = [
  { name: 'You load the page', icon: 0 },
  { name: 'App gets rendered and api call is being made', icon: 0 },
  {
    name: 'Data is fetched from the Api via actions and stored in context via reducer',
    icon: 0,
  },
  {
    name: 'Home Page then renders the cart component ,based on the data from context, where you can add/remove/select items to purchase',
    icon: 0,
  },
  { name: 'Button gets disabled / enabled based on items you select', icon: 0 },
  {
    name: 'On further steps you have option to select address or add a new address',
    icon: 0,
  },
  {
    name: 'You may add a new address which will post be shown in the adress list',
    icon: 0,
  },
  {
    name: 'On next step you will have an option to check the price and adress and submit it',
    icon: 0,
  },
  {
    name: 'On Api success , a success toast will be shown with success message and cart becomes empty',
    icon: 0,
  },
  {
    name: 'On failure you will be at the same page and left with an option to retry',
    icon: 0,
  },
  { name: 'Thats the end :)', icon: 0 },
];
const partners= [
  "Frontend partner: React JS",
  "Styling partner: Tailwind css",
  "Testing partner: React testing library",
  "Testing partner: React testing library",
  "Deployment partner : Vercel"
]
function Arch() {
  return (
    <div className="container mx-auto w-128 ">
      <div className="mt-4">
        <h3 className="mb-2 text-lg font-semibold text-gray-900 dark:text-white">
          Partners in the project
        </h3>
        <ul className="max-w-md space-y-1 text-gray-500 list-inside dark:text-gray-400">
          {
            partners.map(partner=>(
              <li className="flex items-center">
              <Tick />
             {partner}
            </li>
            ))
          }
        </ul>
      </div>
      <div className="mt-4">
        <h3 className="mb-2 text-lg font-semibold text-gray-900 dark:text-white">
          Data Flow
        </h3>
        <div className="flex flex-col space-y-4 p-4">
          {dataFlow.map((data, id) => {
            return (
              <div className="flex font-sm items-center">
                <span className="text-green-500 mr-2">
                  {data.icon === 0 ? '✓' : '•'}
                </span>
                <span className="font-bold">{data.name}</span>
              </div>
            );
          })}
        </div>
      </div>
      <div className="mt-4">
        <h3 className="mb-2 text-lg font-semibold text-gray-900 dark:text-white">
          HLD Diagram
        </h3>
        <img
          src="https://rukminim1.flixcart.com/www/800/800/promos/26/01/2024/04b6c585-1a76-475a-a2ee-34cfe0806242.png?q=100"
          alt="design diagrams"
        />
      </div>
      <div className="mt-4">
        <h3 className="mb-2 text-lg font-semibold text-gray-900 dark:text-white">
          Data Flow in the project
        </h3>
        <img
          src="https://rukminim1.flixcart.com/www/1000/1000/promos/26/01/2024/c43eae7a-3773-4361-9e2f-fb7fbc3be3a9.png?q=100"
          alt="data flow"
        />
      </div>
    </div>
  );
}

export default Arch;
