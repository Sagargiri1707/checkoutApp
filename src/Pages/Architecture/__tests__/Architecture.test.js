import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import Architecture from '../Architecture';

describe('navbar component', () => {
  it('should render a header ', () => {
    const container = render(<Architecture />);
    expect(container).toMatchSnapshot();
  });
});
