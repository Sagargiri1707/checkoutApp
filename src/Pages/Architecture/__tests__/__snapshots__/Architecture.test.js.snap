// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component should render a header  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto w-128 "
      >
        <div
          class="mt-4"
        >
          <h3
            class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
          >
            Partners in the project
          </h3>
          <ul
            class="max-w-md space-y-1 text-gray-500 list-inside dark:text-gray-400"
          >
            <li
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
                />
              </svg>
              Frontend: React JS
            </li>
            <li
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
                />
              </svg>
              Styling: Tailwind css
            </li>
            <li
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
                />
              </svg>
              Testing: React testing library
            </li>
            <li
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
                />
              </svg>
              Testing: React testing library
            </li>
            <li
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
                />
              </svg>
              Deployment : Vercel
            </li>
          </ul>
        </div>
        <div
          class="mt-4"
        >
          <h3
            class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
          >
            Data Flow
          </h3>
          <div
            class="flex flex-col space-y-4 p-4"
          >
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                You load the page
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                App gets rendered and api call is being made
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                Data is fetched from the Api via actions and stored in context via reducer
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                Home Page then renders the cart component ,based on the data from context, where you can add/remove/select items to purchase
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                Button gets disabled / enabled based on items you select
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                On further steps you have option to select address or add a new address
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                You may add a new address which will post be shown in the adress list
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                On next step you will have an option to check the price and adress and submit it
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                On Api success , a success toast will be shown with success message and cart becomes empty
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                On failure you will be at the same page and left with an option to retry
              </span>
            </div>
            <div
              class="flex font-sm items-center"
            >
              <span
                class="text-green-500 mr-2"
              >
                ✓
              </span>
              <span
                class="font-bold"
              >
                Thats the end :)
              </span>
            </div>
          </div>
        </div>
        <div
          class="mt-4"
        >
          <h3
            class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
          >
            HLD Diagram
          </h3>
          <img
            alt="design diagrams"
            src="https://rukminim1.flixcart.com/www/800/800/promos/26/01/2024/04b6c585-1a76-475a-a2ee-34cfe0806242.png?q=100"
          />
        </div>
        <div
          class="mt-4"
        >
          <h3
            class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
          >
            Data Flow in the project
          </h3>
          <img
            alt="data flow"
            src="https://rukminim1.flixcart.com/www/1000/1000/promos/26/01/2024/c43eae7a-3773-4361-9e2f-fb7fbc3be3a9.png?q=100"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto w-128 "
    >
      <div
        class="mt-4"
      >
        <h3
          class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
        >
          Partners in the project
        </h3>
        <ul
          class="max-w-md space-y-1 text-gray-500 list-inside dark:text-gray-400"
        >
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
              />
            </svg>
            Frontend: React JS
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
              />
            </svg>
            Styling: Tailwind css
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
              />
            </svg>
            Testing: React testing library
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
              />
            </svg>
            Testing: React testing library
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="w-3.5 h-3.5 me-2 text-green-500 dark:text-green-400 flex-shrink-0"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 .5a9.5 9.5 0 1 0 9.5 9.5A9.51 9.51 0 0 0 10 .5Zm3.707 8.207-4 4a1 1 0 0 1-1.414 0l-2-2a1 1 0 0 1 1.414-1.414L9 10.586l3.293-3.293a1 1 0 0 1 1.414 1.414Z"
              />
            </svg>
            Deployment : Vercel
          </li>
        </ul>
      </div>
      <div
        class="mt-4"
      >
        <h3
          class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
        >
          Data Flow
        </h3>
        <div
          class="flex flex-col space-y-4 p-4"
        >
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              You load the page
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              App gets rendered and api call is being made
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              Data is fetched from the Api via actions and stored in context via reducer
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              Home Page then renders the cart component ,based on the data from context, where you can add/remove/select items to purchase
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              Button gets disabled / enabled based on items you select
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              On further steps you have option to select address or add a new address
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              You may add a new address which will post be shown in the adress list
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              On next step you will have an option to check the price and adress and submit it
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              On Api success , a success toast will be shown with success message and cart becomes empty
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              On failure you will be at the same page and left with an option to retry
            </span>
          </div>
          <div
            class="flex font-sm items-center"
          >
            <span
              class="text-green-500 mr-2"
            >
              ✓
            </span>
            <span
              class="font-bold"
            >
              Thats the end :)
            </span>
          </div>
        </div>
      </div>
      <div
        class="mt-4"
      >
        <h3
          class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
        >
          HLD Diagram
        </h3>
        <img
          alt="design diagrams"
          src="https://rukminim1.flixcart.com/www/800/800/promos/26/01/2024/04b6c585-1a76-475a-a2ee-34cfe0806242.png?q=100"
        />
      </div>
      <div
        class="mt-4"
      >
        <h3
          class="mb-2 text-lg font-semibold text-gray-900 dark:text-white"
        >
          Data Flow in the project
        </h3>
        <img
          alt="data flow"
          src="https://rukminim1.flixcart.com/www/1000/1000/promos/26/01/2024/c43eae7a-3773-4361-9e2f-fb7fbc3be3a9.png?q=100"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
